Mathlib/Data/Fin/Basic.lean

```diff
@@ -2543,4 +2544,15 @@ protected theorem zero_mul [NeZero n] (k : Fin n) : (0 : Fin n) * k = 0 := by
 
 end Mul
 
+open Qq in
+unsafe instance toExpr (n : ℕ) : Lean.ToExpr (Fin n) where
+  toTypeExpr := q(Fin $n)
+  toExpr := match n with
+    | 0 => finZeroElim
+    | n'@(n + 1) => fun i =>
+      let i' : Q(ℕ) := Lean.ToExpr.toExpr (i : ℕ)
+      let _n_eq : ($n + 1) =Q $n' := ⟨⟩
+      q((haveI : NeZero $n' := @NeZero.succ $n; OfNat.ofNat $i' : Fin $(n')))
+#align fin.reflect Fin.toExprₓ
```

Comment:
@gebner, how can I use `_n_eq` here to avoid `q()` complaining that my types don't match?

Reply:
I have tried my hand at refactoring this instance.